backend/server.go

```diff
@@ -0,0 +1,183 @@
+package backend
+
+import (
+	rice "github.com/GeertJohan/go.rice"
+	"github.com/gomodule/redigo/redis"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type Data struct {
+	People []Person
+	Awards []string
+}
+
+type Person struct {
+	Name string `redis:"name"`
+	Score float64 `redis:"score"`
+}
+
+type App struct {
+	pool *redis.Pool
+	config Config
+	tmpl 	template.Template
+}
+
+func renderTemplate(w http.ResponseWriter, people []Person, tmpl template.Template) {
+	err := tmpl.ExecuteTemplate(w, "index.html", Data{
+		People: people,
+		Awards: awards,
+	})
+	checkError(w, err)
+}
+
+func checkError(w http.ResponseWriter, err error) {
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func (app *App) homeHandler(w http.ResponseWriter, r *http.Request) {
+	people, err := app.loadPeople()
+	checkError(w, err)
+
+	renderTemplate(w, people, app.tmpl)
+
+	awards = make([]string ,0)
+}
+
+func (app *App) addHandler(w http.ResponseWriter, r *http.Request) {
+	conn := app.pool.Get()
+
+	err := r.ParseForm()
+	checkError(w, err)
+
+	for key, values := range r.Form {
+		name := strings.SplitN(key, "-", 2)[1]
+		value := values[0]
+
+		if value == "" {
```

Comment:
What if value is " "? Is there a library that you could use to strip out the beginning and trailing white spaces/blank text

Reply:
This is actually a side-effect of using the very primal <form> submit from frontend. It not only submits the input field with the actual input in it, but also all the other blank fields (e.g. if there are 20 users and I only want to update the score for one of them, it would submit all 20 users with blank string for the 19 of them and the score for the 1 that I want). And for each input field, the value is submitted as an array of strings. So the 19 users would actually have value like [""], and the 1 user would have a value like ["3.1415926"] where 3.1415926 is the amount of points I want to add to them. Thus the condition value == "".